### ISSUE TITLE
Cannot redefine property: usingDomains

### ISSUE THREAD

**Issue by odino** [petkaantonov/bluebird#559]
I am using bluebird through request-promise, just did an npm install and...BOOM :) I think one of the latest versions might have introduced an issue

```
web_1   | /src/node_modules/request-promise/node_modules/bluebird/js/main/async.js:75
web_1   |         Object.defineProperty(EventsModule, "usingDomains", {
web_1   |                ^
web_1   | TypeError: Cannot redefine property: usingDomains
web_1   |     at Function.defineProperty (native)
web_1   |     at Object.<anonymous> (/src/node_modules/request-promise/node_modules/bluebird/js/main/async.js:75:16)
web_1   |     at Module._compile (module.js:456:26)
web_1   |     at Object.Module._extensions..js (module.js:474:10)
web_1   |     at Module.load (module.js:356:32)
web_1   |     at Function.Module._load (module.js:312:12)
web_1   |     at Module.require (module.js:364:17)
web_1   |     at require (module.js:380:17)
web_1   |     at module.exports (/src/node_modules/request-promise/node_modules/bluebird/js/main/promise.js:13:13)
web_1   |     at Object.<anonymous> (/src/node_modules/request-promise/lib/bluebird-fresh.js:2:53)
web_1   |     at Module._compile (module.js:456:26)
web_1   |     at Object.Module._extensions..js (module.js:474:10)
web_1   |     at Module.load (module.js:356:32)
web_1   |     at Function.Module._load (module.js:312:12)
web_1   |     at Module.require (module.js:364:17)
web_1   |     at require (module.js:380:17)
```

**Comment by odino** [comment]
BTW I guess the code that messes up with this is in [here](https://github.com/petkaantonov/bluebird/commit/3e36fc861ac5795193ba37935333eb6ef3716390#diff-21a685e9d171679552accc32acea40ecR73)

Dont ask me why it fails, as that if / else is exactly there to avoid this :-S

**Comment by odino** [comment]
To add on this: just checked and `2.9.22` looks fine

**Comment by petkaantonov** [comment]
Fixed in 2.9.24

**Comment by odino** [comment]
thanks man :)
